chore: cut v0.73.0 (#6468)
prometheus-operator/prometheus-operator: CHANGELOG.md

## 0.73.0 / 2024-04-03

> * [CHANGE/BUGFIX] Add `proxyURL` validation for ServiceMonitor, PodMonitor and Probe objects. #6464
>   *It could be a breaking change for those who rely on the configuration reloader sidecar to expand environment variables in the generated configuration file.*

Reviewer: ```suggestion
* [CHANGE/BUGFIX] Add `proxyURL` validation for ServiceMonitor, PodMonitor and Probe objects. **It could be a breaking change for those who rely on the configuration reloader sidecar to expand environment variables in the generated configuration file.** #6464  
```

Author: done

## 0.73.0 / 2024-04-03

> * [CHANGE] Add proxyURL validation for servicemon,podmonitor and probe #6464

Reviewer: ```suggestion
* [CHANGE/BUGFIX] Add `proxyURL` validation for ServiceMonitor, PodMonitor and Probe objects. #6464
```

>       The fix might be a breaking change for those who rely on the config reloader to expand env variables in the config.

Reviewer: ```suggestion
     *It could be a breaking change for those who rely on the configuration reloader sidecar to expand environment variables in the generated configuration file.*
```

> * [FEATURE] Add summary to msteams #6206

Reviewer: ```suggestion
* [FEATURE] Add `summary` field to the MSteams receiver. #6206
```

> * [FEATURE] Add support for Kuma SD in ScrapeConfig #6427,#6465

Reviewer: ```suggestion
* [FEATURE] Add support for Kuma SD in `ScrapeConfig` CRD. #6427,#6465
```

* [FEATURE] Add support for Eureka SD in ScrapeConfig #6408
* [FEATURE] Add support for Docker SD in ScrapeConfig #6421
* [FEATURE] Add support for Hetzner SD in ScrapeConfig #6436

> * [FEATURE] Add a flag to set node address priority #6377

Reviewer: ```suggestion
* [FEATURE] Add the `--kubelet-node-address-priority`  CLI argument to set the node address priority. #6377
```

> * [FEATURE] Allow addition of extra relabelings to all scrape classes #6379

Reviewer: ```suggestion
* [FEATURE] Add `relabelings` field to scrape classes. #6379
```

> * [FEATURE] Support `body_size_limit` field in ServiceMonitor and PodMonitor #6349

Reviewer: ```suggestion
* [FEATURE] Add `bodySizeLimit` field to the ServiceMonitor and PodMonitor CRDs. #6349
```

> * [FEATURE] Support `sample_age_limit` field in QueueConfig #6326

Reviewer: ```suggestion
* [FEATURE] Add `sampleAgeLimit` field to the remote-write configuration. #6326
```

* [ENHANCEMENT] Verify which CRDs are installed and start the respective controllers only when needed #6351

> * [ENHANCEMENT] Refactored operator's event handler #6416

Reviewer: this can be dropped as there's no user-facing impact.

> * [ENHANCEMENT] Added extra context to warnings #6410

Reviewer: same here

> * [ENHANCEMENT] Refactored TLS config in ServiceMonitor, PodMonitor and Probe #6392

Reviewer: same

> * [ENHANCEMENT] Declare PodMonitor, Probe and Thanos sidecar as officially supported. #6434

Reviewer: Maybe move it as a preamble of the list?

* [ENHANCEMENT] Add checks for selectors in KubernetesSDConfig #6359

> * [BUGFIX] ProxyConfig fields are now reflected in the in API documentation #6401

Reviewer: we usually don't advertise doc changes.

* [BUGFIX] Fix ScrapeConfigs selection issue across different namespaces #6390

> * [BUGFIX] Added check to determine if thanos support the `--prometheus.http-client` flag #6448

Reviewer: ```suggestion
* [BUGFIX] Add check to determine if Thanos supports the `--prometheus.http-client` flag. #6448
```

> * [BUGFIX] Fixed PrometheusAgent reconciliation when the secret/configmap's namespace is different from the PrometheusAgent namespace. #6412

Reviewer: ```suggestion
* [BUGFIX] Fix PrometheusAgent reconciliation when the secret/configmap's namespace is different from the PrometheusAgent namespace. #6412
```

* [BUGFIX] Allow empty separator in relabel config #6425

> * [BUGFIX] Fixed muteTimeIntervals.timeIntervals.months regex pattern #6362

Reviewer: ```suggestion
* [BUGFIX] Fix the validation for `muteTimeIntervals`'s months in the AlertmanagerConfig CRD. #6362
```
